File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    dict = {players[i] : i for i in range(len(players))}
    for c in callings:  
        idx = dict[c]
        dict[c] -= 1
        dict[players[idx - 1]] += 1
        players[idx], players[idx - 1] = players[idx - 1], players[idx]
    
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    var answer = [...players];
    
    const playerList = {}
    for (let i = 0; i < answer.length; i++) {
        playerList[answer[i]] = i;
    }
    
    for (let calling of callings) {
        const index = playerList[calling];
        playerList[calling]--;
        playerList[answer[index - 1]]++;
        
        const tmp = answer[index - 1];
        answer[index - 1] = calling;
        answer[index] = tmp;
    }
    
    return answer;
}